base/src/main/java/org/geysermc/api/util/ApiVersion.java

```diff
@@ -0,0 +1,53 @@
+package org.geysermc.api.util;
+
+/**
+ * Represents a version of the api.
+ */
+public class ApiVersion {
+    private final int major;
+    private final int minor;
+    private final int patch;
+
+    public ApiVersion(int major, int minor, int patch) {
+        this.major = major;
+        this.minor = minor;
+        this.patch = patch;
+    }
+
+    /**
+     * Returns the major version of the api.
+     *
+     * @return the major version
+     */
+    public int major() {
+        return this.major;
+    }
+
+    /**
+     * Returns the minor version of the api.
+     *
+     * @return the minor version
+     */
+    public int minor() {
+        return this.minor;
+    }
+
+    /**
+     * Returns the patch version of the api.
+     *
+     * @return the patch version
+     */
+    public int patch() {
+        return this.patch;
+    }
+
+    /**
+     * Checks whether the given version is compatible with this version.
+     *
+     * @param version the version to check against
+     * @return whether the given version is compatible with this version
+     */
+    public boolean isCompatible(ApiVersion version) {
+        return version.major() != this.major && version.minor() >= minor;
```

Comment:
```suggestion
        return version.major() == this.major && version.minor() >= minor;
```
We want the major versions to match up, xd.

Reply:
Also, its completely ambiguous whether `this` or `version` is the api consumer or provider. Is this method being used correctly in the other PR?  

If it is only geyser/base api that should have instances of ApiVersion, maybe this method should just take major and minor ints. Makes its a bit less error prone, and we don't need to make an ApiVersion while loading an extension just to unwrap it and throw it away.

Reply:
Yeah, in the other PR, the extension version is the parameter

if we want the api consumer (extension) version to be the parameter to this method (which i would like to see), then it would be `consumerMinor <= minor`

since the extensions declare a minimum api minor version it requires, for new features

Reply:
Thanks konica; ill add docs on the parameter & fix the version check.